elm/Main.elm

```diff
@@ -85,11 +59,32 @@ update msg model =
                     |> Maybe.fromJust
 
             in
-               { model | setModels = models }
+               ({ model | setModels = models }, Cmd.none)
+
+        MovementInputMsg msg ->
+            let
+                (updated, cmd) = MovementInput.update msg model.movements
+            in
+                ({ model | movements = updated }, Cmd.map MovementInputMsg cmd)
+
+        AddSet ->
+            let
+                count = 0
+                newSets = model.setModels ++ [SingleSet.initialModel count]
```

Comment:
Is there a way to roll up this into a function on `SingleSet` that captures adding a new set (including no side effects) so the structure would be similar to that of `MovementInputMsg msg`s?

Reply:
Yeah, I'm almost certainly going to move in that direction. Like with the code above, this is an artifact and can be ignored right now.

Reply:
Cool beans